src/DotNetWorker.ApplicationInsights/Initializers/AppServiceOptionsInitializer.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using Microsoft.Azure.Functions.Worker.ApplicationInsights.Initializers;
+using Microsoft.Extensions.Options;
+
+internal class AppServiceOptionsInitializer : IConfigureOptions<AppServiceOptions>
+{
+    internal const string AzureWebsiteName = "WEBSITE_SITE_NAME";
+    internal const string AzureWebsiteSlotName = "WEBSITE_SLOT_NAME";
+    internal const string AzureWebsiteCloudRoleName = "WEBSITE_CLOUD_ROLENAME";
+    internal const string DefaultProductionSlotName = "production";
+
+    internal static string[] EnvironmentVariablesToMonitor = [AzureWebsiteName, AzureWebsiteSlotName, AzureWebsiteCloudRoleName];
+
+    public void Configure(AppServiceOptions options)
+    {
+        options.AzureWebsiteName = Environment.GetEnvironmentVariable(AzureWebsiteName);
```

Comment:
Should we consider using `IConfiguration` to read this value? Is `IConfiguration` available at this point?

Reply:
`IConfiguration` caches all the environment variables so unfortunately this doesn't really work as it'd never see a change. We *could* write a broader environment variable checker that'd invalidate the `IConfiguration` and have it re-populate if there's a change, but that's a much bigger issue and I don't think it's worth it. In this case, we only really care about a couple platform-level variables, so I elected to just check them directly. 

App Insights takes a similar approach here -- https://github.com/microsoft/ApplicationInsights-dotnet/blob/main/WEB/Src/WindowsServer/WindowsServer/Implementation/EnvironmentVariableMonitor.cs